test/v2/controllers/V2MovementsControllerSpec.scala

```diff
@@ -1545,82 +1595,229 @@ class V2MovementsControllerSpec
-              any[HeaderCarrier],
-              any[ExecutionContext]
+          when(mockPushNotificationService.associate(any[String].asInstanceOf[MovementId], any[MovementType], any())(any(), any()))
+            .thenAnswer(
+              _ => EitherT.rightT(boxResponse)
             )
-        )
-          .thenAnswer(
-            _ =>
-              EitherT.rightT(
-                MovementResponse(MovementId("123"), Some(MessageId("456")))
-              )
+
+          when(
+            mockRouterService.send(
+              eqTo(MessageType.ArrivalNotification),
+              any[String].asInstanceOf[EORINumber],
+              any[String].asInstanceOf[MovementId],
+              any[String].asInstanceOf[MessageId],
+              any[Source[ByteString, _]]
+            )(any[ExecutionContext], any[HeaderCarrier])
           )
+            .thenAnswer {
+              _ =>
+                EitherT.leftT(RouterError.UnexpectedError(None))
+            }
 
-        when(
-          mockRouterService.send(
+          val request = fakeCreateMovementRequest("POST", standardHeaders, singleUseStringSource(CC007Cjson), MovementType.Arrival)
+
+          val result = sut.createMovement(MovementType.Arrival)(request)
+          status(result) mustBe INTERNAL_SERVER_ERROR
+
+          verify(mockValidationService, times(1)).validateJson(eqTo(MessageType.ArrivalNotification), any())(any(), any())
+          verify(mockConversionService).jsonToXml(eqTo(MessageType.ArrivalNotification), any())(any(), any(), any())
+          verify(mockMovementsPersistenceService).createMovement(any[String].asInstanceOf[EORINumber], any[MovementType], any())(any(), any())
+          verify(mockRouterService).send(
             eqTo(MessageType.ArrivalNotification),
             any[String].asInstanceOf[EORINumber],
             any[String].asInstanceOf[MovementId],
             any[String].asInstanceOf[MessageId],
             any[Source[ByteString, _]]
           )(any[ExecutionContext], any[HeaderCarrier])
+      }
+
+    }
+
+    "when the content type is not present" - {
+
+      val standardHeaders = FakeHeaders(
+        Seq(HeaderNames.ACCEPT -> "application/vnd.hmrc.2.0+json")
+      )
+
+      "must return Accepted if the call to upscan and the persistence service succeeds" in forAll(
+        arbitraryUpscanInitiateResponse.arbitrary,
+        arbitraryMovementResponse(false).arbitrary,
+        arbitraryBoxResponse.arbitrary
+      ) {
+        (upscanResponse, movementResponse, boxResponse) =>
+          beforeEach()
+
+          when(mockUpscanService.upscanInitiate(any[String].asInstanceOf[MovementId])(any(), any()))
+            .thenAnswer {
+              _ => EitherT.rightT(upscanResponse)
+            }
+
+          when(mockAuditService.audit(any(), any(), eqTo(MimeTypes.XML))(any(), any())).thenReturn(Future.successful(()))
```

Comment:
we can remove this mock, as we are not sending any audit message

Reply:
Good point!